lib/galaxy/managers/groups.py

```diff
@@ -83,32 +71,48 @@ def show(self, trans: ProvidesAppContext, group_id: int):
         encoded_id = DecodedDatabaseIdField.encode(group_id)
         group = self._get_group(trans.sa_session, group_id)
         item = group.to_dict(view="element", value_mapper={"id": DecodedDatabaseIdField.encode})
-        item["url"] = url_for("group", id=encoded_id)
-        item["users_url"] = url_for("group_users", group_id=encoded_id)
-        item["roles_url"] = url_for("group_roles", group_id=encoded_id)
+        item["url"] = self._url_for(trans, "show", group_id=encoded_id)
+        encoded_user_id = [DecodedDatabaseIdField.encode(gu.user.id) for gu in group.users]
+        encoded_role_id = [DecodedDatabaseIdField.encode(gr.role.id) for gr in group.roles]
+        item["users_url"] = [
+            self._url_for(trans, "group_user", group_id=encoded_id, user_id=group_user)
+            for group_user in encoded_user_id
+        ]
+        item["roles_url"] = [
+            self._url_for(trans, "group_role", group_id=encoded_id, role_id=group_role)
+            for group_role in encoded_role_id
+        ]
```

Comment:
The same for these URLs, they should not be lists but rather something like:
`users_url: "/api/groups/{encoded_group_id}/users"`

I also think some of those URLs are currently wrong in the `dev` branch which kind of confirms that nobody is using them... so we should probably deprecate them and remove them at some point.